lib/core/public/configure.js

```diff
@@ -45,14 +45,37 @@ function configure(spec) {
 	}
 
 	if (spec.checks) {
+		if (!Array.isArray(spec.checks)) {
+			throw new TypeError('Checks property must be an array');
+		}
+
 		spec.checks.forEach(function(check) {
+			if (!check.id) {
+				const checkStr =
+					typeof check === 'object' ? JSON.stringify(check) : check;
+				throw new TypeError(
+					`Configured check "${checkStr}" is invalid. Checks must be an object with at least an id property`
+				);
```

Comment:
NBD, but we don't need the above ternary. Can just always stringify the "check":

```suggestion
				throw new TypeError(
					`Configured check ${JSON.stringify(check)} is invalid. Checks must be an object with at least an id property`
				);
```